src/jdmtool/data_card/garmin.py

```diff
@@ -19,38 +22,76 @@ class GarminFirmwareWriter(BasicUsbDevice):
     WRITE_ENDPOINT = -1
     READ_ENDPOINT = -1
 
+    def write_firmware_0x300(self) -> None:
+        import time
+        print("Writing 0x300 part 1 of 2")
+        with open(FIRMWARE_DIR / 'grmn0300-part1.dat', 'rb') as fd:
+            self.write_firmware(fd)
+        time.sleep(2)
+        print("Writing 0x300 part 2 of 2")
+        with open(FIRMWARE_DIR / 'grmn0300-part2.dat', 'rb') as fd:
+            self.write_firmware(fd)
+        time.sleep(2)
+
```

Comment:
Interesting... Does it need to be written in two parts, with a 2s pause? What happens otherwise?

There are existing opensource tools for writing the firmware - e.g., fxload (https://github.com/mbed-ce/fxload/). It doesn't seem to have any sleep statements. But it _does_ have a retry logic for each control write - I wonder if retrying is enough?

(I've been meaning to switch from my custom .dat format into .ihex... Then, in theory, you could literally use fxload to write the firmware. But I originally wrote all of this code before I discovered fxload.)